examples/tree_1d_dgsem/elixir_shallowwater_well_balanced_wet_dry.jl

```diff
@@ -0,0 +1,183 @@
+# ODE solvers, callbacks etc.
+
+tspan = (0.0, 25.0)
+ode = semidiscretize(semi, tspan)
+
+function initial_condition_discontinuous_well_balancedness(x, t, element_id, equations:: ShallowWaterEquations1D)
+  v = 0.0
+  b = sin(4 * pi * x[1]) + 1
+  H = max(b, 1.5)
+
+  # for refinement_level=6
+  if (element_id >= 1 && element_id <= 32)
+    b = sin(4 * pi * x[1]) + 3
+    H = max(b, 2.5)
+  end
+
+  # Shift the water level by the amount `equations.threshold_limiter` (1e-13 per default)
+  # to avoid division by a hard 0 value in the water height `h` when computing velocities.
+  H = max(H, b + equations.threshold_limiter)
+  return prim2cons(SVector(H, v, b), equations)
+end
+
+# point to the data we want to augment
+u = Trixi.wrap_array(ode.u0, semi)
+# reset the initial condition
+for element in eachelement(semi.solver, semi.cache)
+  for i in eachnode(semi.solver)
+    x_node = Trixi.get_node_coords(semi.cache.elements.node_coordinates, equations, semi.solver, i, element)
+    u_node = initial_condition_discontinuous_well_balancedness(x_node, first(tspan), element, equations)
+    Trixi.set_node_vars!(u, u_node, equations, semi.solver, i, element)
+  end
+end
+
+summary_callback = SummaryCallback()
+
+analysis_interval = 1000
+analysis_callback = AnalysisCallback(semi, interval=analysis_interval, save_analysis=false)
+
+alive_callback = AliveCallback(analysis_interval=analysis_interval)
+
+save_solution = SaveSolutionCallback(interval=5000,
+                                     save_initial_solution=true,
+                                     save_final_solution=true)
+
+stepsize_callback = StepsizeCallback(cfl=1.5)                                     
+
+callbacks = CallbackSet(summary_callback, analysis_callback, alive_callback, save_solution,
+                        stepsize_callback)
+
+stage_limiter! = PositivityPreservingLimiterShallowWater(thresholds=(equations.threshold_limiter,),
+                                                     variables=(Trixi.waterheight,))
+
+###############################################################################
+# run the simulation
+
+sol = solve(ode, SSPRK43(stage_limiter!), dt=1.0,
+            save_everystep=false, callback=callbacks, adaptive=false);
+
+summary_callback() # print the timer summary
+
+###############################################################################
+# workaround to compute the well-balancedness error for this partiular problem
+# that has two reference water heights. One for a lake to the left of the
+# discontinuous bottom topography `H0_upper = 2.5` and another for a lake to the
+# right of the discontinuous bottom topography `H0_lower = 1.5`.
+
+# Declare a special version of the function to compute the lake-at-rest error
+# OBS! The reference water height values are hardcoded for convenience.
+function lake_at_rest_error_two_level(u, element_id, equations::ShallowWaterEquations1D)
```

Comment:
Similar here. this function can use the perturbed node value for `x` instead of the element ID to set the discontinuity.